lib/pperm.gi

```diff
@@ -225,7 +226,8 @@ InstallMethod(AsPartialPerm, "for a transformation and list",
 [IsTransformation, IsList], 
 function(f, list)
   
-  if not IsSSortedList(list) or not ForAll(list, IsPosInt) then 
+  if not IsSSortedList(list)
+      or not ForAll(list, x -> IsSmallIntRep(x) and IsPosInt(x)) then 
```

Comment:
Off-topic: you can actually write `ForAll(list`, IsSmallIntRep and IsPosInt);` (whether that's a good idea or not is probably a matter of test; in any case, I am not saying you should change this, just pointing it out ;-)